os-server/src/main/java/app/openschool/course/CourseServiceImpl.java

```diff
@@ -30,4 +66,129 @@ public Page<Course> findAll(
     return courseRepository.findAll(
         courseTitle, subCategoryIds, languageIds, difficultyIds, pageable);
   }
+
+  @Override
+  public Course add(CreateCourseRequest request) {
+    Course course = new Course();
+    course.setTitle(request.getTitle());
+    course.setDescription(request.getDescription());
+    course.setGoal(request.getGoal());
+    course.setCategory(
+        categoryRepository
+            .findById(request.getCategoryId())
+            .orElseThrow(IllegalArgumentException::new));
+    course.setDifficulty(
+        difficultyRepository
+            .findById(request.getDifficultyId())
+            .orElseThrow(IllegalArgumentException::new));
+    course.setLanguage(
+        languageRepository
+            .findById(request.getLanguageId())
+            .orElseThrow(IllegalArgumentException::new));
+    course.setMentor(
+        userRepository.findUserByEmail(
+            SecurityContextHolder.getContext().getAuthentication().getName()));
+    Set<Keyword> keywords =
+        request.getKeywordIds().stream()
+            .map(
+                keywordId ->
+                    keywordRepository
+                        .findById(keywordId)
+                        .orElseThrow(IllegalArgumentException::new))
+            .collect(Collectors.toSet());
+    course.getKeywords().addAll(keywords);
+    course.setModules(createModules(request.getCreateModuleRequests(), course));
+    return courseRepository.save(course);
+  }
+
+  private Set<Module> createModules(Set<CreateModuleRequest> requests, Course course) {
```

Comment:
this method should return created modules then outside the method you can add them to the course